marathon_acme/tests/test_sse_protocol.py

```diff
@@ -13,254 +17,326 @@ def loseConnection(self):
+        assert_that(protocol.transport.disconnecting, Is(True))
 
-    def test_transport_disconnecting(self):
+    def test_transport_disconnecting(self, protocol, messages):
         """
         When the transport for the protocol is disconnecting, processing should
         be halted.
         """
-        self.transport.disconnecting = True
-        self.protocol.dataReceived(b'data:hello\r\n\r\n')
+        protocol.transport.disconnecting = True
+        protocol.dataReceived(b'data:hello\r\n\r\n')
 
-        assert_that(self.messages, Equals([]))
+        assert_that(messages, Equals([]))
 
-    def test_transport_connection_lost(self):
+    def test_transport_connection_lost(self, protocol):
         """
         When the connection is lost, the finished deferred should be called.
         """
-        finished = self.protocol.when_finished()
+        finished = protocol.when_finished()
 
-        self.protocol.connectionLost()
+        protocol.connectionLost()
 
         assert_that(finished, succeeded(Is(None)))
 
-    def test_transport_connection_lost_no_callback(self):
+    def test_transport_connection_lost_no_callback(self, protocol):
         """
         When the connection is lost and the finished deferred hasn't been set,
         nothing should happen.
         """
-        self.protocol.connectionLost()
+        protocol.connectionLost()
 
-    def test_multiple_events_resets_the_event_type(self):
+    def test_multiple_events_resets_the_event_type(self, protocol, messages):
         """
         After an event is consumed with a custom event type, the event type
         should be reset to the default, and the handler should receive further
         messages with the default event type.
         """
         # Event 1
-        self.protocol.dataReceived(b'event:status\r\n')
-        self.protocol.dataReceived(b'data:hello\r\n')
-        self.protocol.dataReceived(b'\r\n')
+        protocol.dataReceived(b'event:status\r\n')
+        protocol.dataReceived(b'data:hello\r\n')
+        protocol.dataReceived(b'\r\n')
 
         # Event 2
-        self.protocol.dataReceived(b'data:world\r\n')
-        self.protocol.dataReceived(b'\r\n')
+        protocol.dataReceived(b'data:world\r\n')
+        protocol.dataReceived(b'\r\n')
 
-        assert_that(self.messages, Equals([
+        assert_that(messages, Equals([
             ('status', 'hello'),
             ('message', 'world')
         ]))
+
+    def test_timeout(self):
+        """
+        When a timeout value is set, the timeout should be triggered once the
+        timeout has elapsed and should disconnect the transport.
+        """
+        timeout = 5
+        clock = Clock()
+        protocol = make_protocol(timeout=timeout, reactor=clock)
+        protocol.connectionMade()
+
+        [delayed_call] = clock.getDelayedCalls()
+        assert_that(delayed_call.getTime(), Equals(clock.seconds() + timeout))
+        assert_that(protocol.transport.disconnecting, Is(False))
```

Comment:
I think we can skip the assertions on the `DelayedCall`. We care about the results, not the implementation details.